test/system/notifier-rewards.test.js

```diff
@@ -0,0 +1,285 @@
+  })
+
+  describe("when notifier notifies about deposit liquidation", () => {
+    beforeEach(async () => {
+      await depositStub.notifyUndercollateralizedLiquidation()
+    })
+
+    context("when both reward amount and percentage are zero", () => {
+      beforeEach(async () => {
+        await riskManagerV1
+          .connect(notifier)
+          .notifyLiquidation(depositStub.address)
+      })
+
+      it("should not be rewarded with asset pool shares", async () => {
+        expect(await underwriterToken.balanceOf(notifier.address)).to.be.equal(
+          0
+        )
+      })
+    })
+
+    context("when reward amount is set but percentage is zero", () => {
+      beforeEach(async () => {
+        await riskManagerV1.beginLiquidationNotifierRewardAmountUpdate(
+          liquidationNotifierRewardAmount
+        )
+        await increaseTime(43200)
+        await riskManagerV1.finalizeLiquidationNotifierRewardAmountUpdate()
+
+        await riskManagerV1
+          .connect(notifier)
+          .notifyLiquidation(depositStub.address)
+      })
+
+      it("should be rewarded with fixed amount of asset pool shares", async () => {
+        expect(await underwriterToken.balanceOf(notifier.address)).to.be.equal(
+          liquidationNotifierRewardAmount
+        )
+      })
+    })
+
+    context("when both reward amount and percentage are set", () => {
+      beforeEach(async () => {
+        await riskManagerV1.beginLiquidationNotifierRewardAmountUpdate(
+          liquidationNotifierRewardAmount
+        )
+        await riskManagerV1.beginLiquidationNotifierRewardPercentageUpdate(
+          liquidationNotifierRewardPercentage
+        )
+        await increaseTime(43200)
+        await riskManagerV1.finalizeLiquidationNotifierRewardAmountUpdate()
+        await riskManagerV1.finalizeLiquidationNotifierRewardPercentageUpdate()
+
+        await riskManagerV1
+          .connect(notifier)
+          .notifyLiquidation(depositStub.address)
+      })
+
+      it("should be rewarded with fixed amount of asset pool shares", async () => {
+        expect(await underwriterToken.balanceOf(notifier.address)).to.be.equal(
+          liquidationNotifierRewardAmount
+        )
+      })
+    })
+
+    context("when reward amount is zero but percentage is set", () => {
+      beforeEach(async () => {
+        await riskManagerV1.beginLiquidationNotifierRewardPercentageUpdate(
+          liquidationNotifierRewardPercentage
+        )
+        await increaseTime(43200)
+        await riskManagerV1.finalizeLiquidationNotifierRewardPercentageUpdate()
+
+        await riskManagerV1
+          .connect(notifier)
+          .notifyLiquidation(depositStub.address)
+      })
+
+      it("should be rewarded with percentage-based amount of asset pool shares", async () => {
+        expect(await underwriterToken.balanceOf(notifier.address)).to.be.equal(
+          covTotalSupply.div(50) // liquidationNotifierRewardPercentage is 2%
```

Comment:
There are no deposits so this is... 0? 

I would rather used a fixed value here instead of computing it so that we can avoid this kind of problems.

Reply:
Nope, the COV supply is minted at the beginning. I agree fixed amount will be better. See https://github.com/keep-network/coverage-pools/pull/111/commits/d1e910303abbb5ac89fe32c0ab6d267a08aa99ce